1-Draft/RFC00XX-PredIctive-IntelliSense.md

```diff
@@ -0,0 +1,359 @@
+
+Predictive IntelliSense displays a best match command completion from command history. The
+prediction may change if a better match is found as the user types additional information. The
+predicted command may be accepted, edited, or ignored.
+
+![Prediction - Historical](./media/pre-wlc.png)
+
+### Install and Remove Predictive IntelliSense
+
+Predictive IntelliSense is a part of the PSReadLine module. Updating or installing the current beta
+release from PSGallery will provide the Predictive IntelliSense feature.
+
+The current release is PSReadLine 2.1.0 Beta 2:
+
+```powershell
+Install-Module PSReadLine -RequiredVersion 2.1.0-beta2 -AllowPrerelease
+```
+
+It is possible to remove, delete, the PSReadLine module to disable Predictive IntelliSense. This
+creates a negative impact to customers as it would remove all features of PSReadLine. For this
+reason, Predictive IntelliSense has its own PSReadLine configuration to enable and disable
+predictions.
+
+### Enabling and Disabling Predictions
+
+By default, the feature Predictive IntelliSense is disabled. The impact of enabling this feature by
+default may cause new and existing users to become confused at the shell prompt. The feature is
+enabled and disabled with a PSReadLine command Set-PSReadLineOption that is executed at the shell
+prompt or in the user’s Profile.
+
+Predictive IntelliSense currently supports four arguments for a prediction source;
+
+* None - This option disables Predictive IntelliSense
+* History - This option uses only the PSReadLine history for predictions
+* Plugin – This option uses only registered plugin modules for predictions
+* HistoryAndPlugin – This option uses both for predictions
+
+To enable Predictive IntelliSense, enter the following command in the shell or in the users Profile:
+
+```powershell
+Set-PSReadLineOption -PredictionSource History
+```
+
+To disable Predictive IntelliSense, enter the following command in the shell or in the users Profile:
+
+```powershell
+Set-PSReadLineOption -PredictionSource None
+```
+
+### Changing the Prediction View
+
+Predictions are displayed in one of two view depending on the user preference. The default view is InlineView.
+
+* InlineView – This is the default view and displays the prediction inline with the user’s typing.
+  This view is similar to other shells Fish and ZSH.
+* ListView – ListView provides a dropdown list of predictions below the line the user is typing.
+  Users may quickly scan the list, highlight and select the desired prediction.
+
+Users may change the view at the command line, or in the user’s PowerShell profile.
+
+```powershell
+Set-PSReadLineOption -PredictionViewStyle InlineView
+```
+
+### Change the Prediction Color for Accessibility
+
+By default, Predictions appear on the same line the user is typing in a different customizable
+color. To support Accessibility needs, the prediction color is settable in the shell or users
+Profile.
+
+![pi-color](./media/pi-color.png)
```

Comment:
The Chrome browser has an interesting color scheme for predicting the URL in the address bar where the predicted completion is highlighted vs greyed out.  I believe the current `-Colors` may already allow specifying both foreground and background color, but maybe we just need an example in the help showing how to set it to that color scheme as we've heard from multiple users that the default grey text is hard to read.

Reply:
@SteveL-MSFT I've seen the background highlighting on the AZ predictor - will get color code and add an example for discussion.